File: 62-unique-paths/unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp=[[1]*n for i in range(m)]
        for i,j in product(range(1,m), range(1,n)):
            dp[i][j]= dp[i-1][j] +dp[i][j-1]
        return dp[m-1][n-1]

        

File: 62-unique-paths/unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<int> dp(n,0);
        
        for(int i=0;i<m;++i)
        {   vector<int> col(n,0);
            for(int j=0;j<n;++j)
            {
                if(i==0 && j==0) col[j]=1;
                else{   int up=0,left=0;
                   if(i>0) up=dp[j];
                   if(j>0) left=col[j-1];
                   col[j]=up+left;
                }
            }dp=col;
        }return dp[n-1];
        
    }

    
};